hivemq-edge/src/frontend/src/extensions/datahub/designer/operation/OperationNode.utils.ts

```diff
@@ -0,0 +1,168 @@
+import { getConnectedEdges, getIncomers, Node } from 'reactflow'
+
+import { PolicyOperation, Schema, Script } from '@/api/__generated__'
+import {
+  DataHubNodeType,
+  DataPolicyData,
+  DryRunResults,
+  OperationData,
+  TransitionData,
+  WorkspaceState,
+} from '@datahub/types.ts'
+import { checkValidityJSScript } from '@datahub/designer/script/FunctionNode.utils.ts'
+import { checkValiditySchema } from '@datahub/designer/schema/SchemaNode.utils.ts'
+import { PolicyCheckErrors } from '@datahub/designer/validation.errors.ts'
+import { isFunctionNodeType, isSchemaNodeType } from '@datahub/utils/node.utils.ts'
+
+export function checkValidityTransformFunction(
+  operationNode: Node<OperationData>,
+  store: WorkspaceState
+): DryRunResults<PolicyOperation, Script | Schema>[] {
+  const { nodes, edges } = store
+
+  if (!operationNode.data.functionId || !operationNode.data.formData) {
+    return [
+      {
+        node: operationNode,
+        error: PolicyCheckErrors.notConfigured(operationNode, 'functionId, formData'),
+      },
+    ]
+  }
+
+  ///////// Check the function handle
+  const functions = getIncomers(operationNode, nodes, edges).filter(isFunctionNodeType)
+
+  if (!functions.length) {
+    return [
+      {
+        node: operationNode,
+        error: PolicyCheckErrors.notConnected(DataHubNodeType.FUNCTION, operationNode),
+      },
+    ]
+  }
+
+  ///////// Check the serializers
+  const serialisers = getIncomers(operationNode, nodes, edges).filter(isSchemaNodeType)
+
+  const connectedEdges = getConnectedEdges([...serialisers], edges).filter(
+    (e) => e.targetHandle === OperationData.Handle.SERIALISER || e.targetHandle === OperationData.Handle.DESERIALISER
+  )
+  const [serial, ...restSerial] = connectedEdges.filter((edge) => edge.targetHandle === OperationData.Handle.SERIALISER)
+  const [deserial, ...restDeserial] = connectedEdges.filter(
+    (edge) => edge.targetHandle === OperationData.Handle.DESERIALISER
+  )
+
+  if (serial === undefined || restSerial.length !== 0) {
+    return [
+      {
+        node: operationNode,
+        error: PolicyCheckErrors.notConnected(DataHubNodeType.SCHEMA, operationNode, OperationData.Handle.SERIALISER),
+      },
+    ]
+  }
+
+  if (deserial === undefined || restDeserial.length !== 0) {
+    return [
+      {
+        node: operationNode,
+        error: PolicyCheckErrors.notConnected(DataHubNodeType.SCHEMA, operationNode, OperationData.Handle.DESERIALISER),
+      },
+    ]
+  }
+
+  // TODO[19240] Should serial and deserial be different ?
+
+  ///////// Check the resources
+  const scriptNodes = functions.map((e) => checkValidityJSScript(e))
+  const schemaNodes = serialisers.map((e) => checkValiditySchema(e))
```

Comment:
suggestion: `e` is very generic while trying to get back into the gist. Let's name the variables by what they represent

```suggestion
  const scriptNodes = functions.map((scriptFunction) => checkValidityJSScript(scriptFunction))
  const schemaNodes = serialisers.map((serialiser) => checkValiditySchema(serialiser))
```